### ISSUE TITLE
[Feature]: Min Soc for selling to grid

### ISSUE THREAD

**Issue by KajLehtinen** [victronenergy/dynamic-ess#75]
### Contact Details

kaj261@gmail.com

### Description

Hi!

It would be nice if a option to define 'min soc to sell to grid' was implemented, that way the system can be configured to only sell to Grid when the batteries is charged up, and it gives the option to the operator of the system to define different values for summer and winter. 

/Kaj 

### Suggested Solution

.

**Comment by fofo444** [comment]
I have a similar request. [Feature]: Planned battery SOC at evening. #70

**Comment by dirkjanfaber** [comment]
Just to let you guys know that we are all seeing the features, but are not responding to them yet. 

The current focus is to get everything stable and running in VRM first. Once that is done, we will start working on adding features.

**Comment by disaster123** [comment]
@dirkjanfaber i just saw that there is already a setting in the remote console:

![image](https://github.com/victronenergy/dynamic-ess/assets/861558/46188b93-6eb0-4910-8489-b50e07abcfc4)

isn't it?

**Comment by dirkjanfaber** [comment]
Yes, that works too. If you prefer to do that from Node-RED, you can do that using the custom output node (copy/paste this via the import menu option in Node-RED):  
```
[{"id":"c68ae0135adcce51","type":"victron-output-custom","z":"287d3e7906396df3","service":"com.victronenergy.settings","path":"/Settings/DynamicEss/MinSoc","serviceObj":{"service":"com.victronenergy.settings","name":"com.victronenergy.settings"},"pathObj":{"path":"/Settings/DynamicEss/MinSoc","name":"/Settings/DynamicEss/MinSoc","type":"number"},"name":"","onlyChanges":false,"x":700,"y":940,"wires":[]},{"id":"f6aace05b3c8309e","type":"inject","z":"287d3e7906396df3","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"10","payloadType":"num","x":310,"y":920,"wires":[["c68ae0135adcce51"]]},{"id":"190cfcf1d5c4cc55","type":"inject","z":"287d3e7906396df3","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"20","payloadType":"num","x":310,"y":960,"wires":[["c68ae0135adcce51"]]}]
```

Combine this with a [bigtimer](https://flows.nodered.org/node/node-red-contrib-bigtimer) or so to adjust for summer or winter.

**Comment by KajLehtinen** [comment]
Hi!

If setting it from within the remote console under ESS/DESS, will the Node-red implementation 'know' about the minimum DESS Soc? Currently it doesn't seem so for me atleast. 

Last night DESS planned that my install, around 10am this morning would be down to approx 25% Soc, but overnight it has consumed from grid approx 3.8 Kwh so the Soc is around 45% at 7am, and DESS changed it plans so that it will sell to grid between 8-11 am. But now when its 8.30 it still hasnt started selling which I contribute to that I set the Minimum DESS SOC in Remote Console to 50% last night.
![image](https://github.com/victronenergy/dynamic-ess/assets/102818048/c62068c8-4531-4932-a107-b12af55edf71)

/Kaj
